documentation/specs/Banner.md

```diff
@@ -0,0 +1,46 @@
+# Banner Component Specification
+
+## Overview
+
+The `Banner` component displays a prominent message above the navigation bar at the top of the page that cannot be dismissed.
+
+### Use Cases
+
+The primary use case of the `Banner` component is to inform the user of promotions and other offerings on the marketing (logged out) site. **Not** to be used to call attention to the user as a result of interacting with the UI.
+
+---
+
+## Design
+
+### API
+
+```typescript
+export type BannerColor = "primary" | "success" | "neutral";
+export type BannerProps = {
+  /**
+   * Banner color
+   * @default "primary"
+   */
+  color?: BannerColor;
+  /**
+   * Content to display
+   */
+  children: ReactNode;
+};
+```
+
+### Example Usage
+
+```tsx
+import Banner from "@easypost/easy-ui/Banner";
+import Text from "@easypost/easy-ui/Text";
+
+<Banner color="success">
+  <Text variant="subtitle1">
+    Limited Time Only:
+  <Text>
+  <Text variant="body1">
+    Get your first $250 of labels free!
+  <Text>
+</Banner>
+```
```

Comment:
my thinking for this component was that we would use it in conjunction with the `Text` component as opposed to accepting props like boldText, bodyText..etc

Reply:
this is a good question. i like your thought process in general but i feel a little uncertain.

looking at Figma, it almost looks like the "subtitle" is treated as a special thing in the comment. for instance, i notice that on mobile screens it breaks the lines into two, which wouldn't be easy to handle on the consumer side with just two `<Text />` components as the content:

<img width="377" alt="image" src="https://user-images.githubusercontent.com/752942/231207667-30123a1a-0ea1-4802-8771-700aefad1b60.png">

it may be worth asking design if the "subtitle" is meant to be an intentional pattern that we should codify in our component API. if it isn't, it would be worth giving thought as to how we would handle the mobile situation where it breaks into two lines

Reply:
good callout, i checked with design and it is indeed an intentional pattern for the component to be able to render semibold and body text in which the semibold text stacks on the body text on mobile devices. for cases when there is no semibold text, the body text should just wrap.

with that said, im curious how do you feel about the updated component API. per design, its a valid use case for either (`title` and `body`) to be able to handle a text link. otherwise i would've just made `title` a string and deferred the `body` to `children`. of course, we can still do something similar,

### API

```typescript
export type BannerStatus = "primary" | "success" | "neutral";
export type BannerProps = {
  /**
   * Banner status color
   * @default "success"
   */
  status?: BannerColor;
  /**
   * Banner title text
   * @default null
   */
  title?: Text;
  /**
   * Banner body content
   */
  children: ReactNode;
};
```
### Usage

```tsx
import Banner from "@easypost/easy-ui/Banner";
import Text from "@easypost/easy-ui/Text";

<Banner title={<Text>Limited Time Only:</Text>}>
Get your first $250 of labels free!
</Banner>
;
```

maybe thats a bit better, curious how you feel about it though.

Reply:
sounds good, updated spec to reflect pattern with using `children`. i also updated the previous `title` prop to `emphasis`. i feel the same about coming up with a name for that one lol, but i think `emphasis` is a bit more contextually appropriate.